In [1]:
import numpy as np

# Given box vectors
box_vectors = np.array([
    [2, 1],  # top left box
    [1, 1],  # top right box
    [6, 3],  # bottom left box
    [3, 3]   # bottom right box
])

# Computing the correlation matrix
correlation_matrix = np.corrcoef(box_vectors, rowvar=False)
correlation_matrix

array([[1.        , 0.80178373],
       [0.80178373, 1.        ]])

Correlation 0.80178373 shows strong correlation.
We will consider the ratio of the boxes

In [3]:
ratios = [box[0] / box[1] for box in box_vectors]
box_ratios = list(zip(box_vectors, ratios))
box_ratios

[(array([2, 1]), 2.0),
 (array([1, 1]), 1.0),
 (array([6, 3]), 2.0),
 (array([3, 3]), 1.0)]